### PR TITLE
[Modules] Created module MaintenanceConfiguration & added missing closing dots for metadata

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Created module MaintenanceConfiguration
- Added missing closing dots for metadata

| Pipeline |
| - |
| [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=117&branchName=users%2Falsehr%2FmaintenanceConfig) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Maintenance/maintenanceConfigurations/.bicep/nested_roleAssignments.bicep [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+@sys.description('Optional. Version of the condition.')
+@allowed([
+  '2.0'
+])
+param conditionVersion string = '2.0'
+
+@sys.description('Optional. Id of the delegated managed identity resource.')
+param delegatedManagedIdentityResourceId string = ''
+
+var builtInRoleNames = {
```
Comment (eriqua) [inline modules/Microsoft.Maintenance/maintenanceConfigurations/.bicep/nested_roleAssignments.bicep:36]: To be updated with the new approach (no spaces, no quotes)

File: modules/Microsoft.Maintenance/maintenanceConfigurations/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    delegatedManagedIdentityResourceId: contains(roleAssignment, 'delegatedManagedIdentityResourceId') ? roleAssignment.delegatedManagedIdentityResourceId : ''
+    resourceId: maintenanceConfiguration.id
+  }
+}]
+
+// =========== //
+//   Outputs   //
+// =========== //
+
+@description('The name of the maintenanceConfiguration.')
```
Comment (eriqua) [inline modules/Microsoft.Maintenance/maintenanceConfigurations/deploy.bicep:111]:
```suggestion
@description('The name of the Maintenance Configuration.')
```

File: modules/Microsoft.Maintenance/maintenanceConfigurations/deploy.bicep [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+output name string = maintenanceConfiguration.name
+
+@description('The resource ID of the maintenanceConfiguration.')
```
Comment (eriqua) [inline modules/Microsoft.Maintenance/maintenanceConfigurations/deploy.bicep:114]:
```suggestion
@description('The resource ID of the Maintenance Configuration.')
```

File: modules/Microsoft.Maintenance/maintenanceConfigurations/deploy.bicep [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+output resourceId string = maintenanceConfiguration.id
+
+@description('The name of the resource group the maintenanceConfiguration was created in.')
```
Comment (eriqua) [inline modules/Microsoft.Maintenance/maintenanceConfigurations/deploy.bicep:117]:
```suggestion
@description('The name of the resource group the Maintenance Configuration was created in.')
```

File: modules/Microsoft.Maintenance/maintenanceConfigurations/deploy.bicep [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+output resourceGroupName string = resourceGroup().name
+
+@description('The location the maintenanceConfiguration was created in.')
```
Comment (eriqua) [inline modules/Microsoft.Maintenance/maintenanceConfigurations/deploy.bicep:120]:
```suggestion
@description('The location the Maintenance Configuration was created in.')
```

File: modules/Microsoft.Maintenance/maintenanceConfigurations/.bicep/nested_roleAssignments.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply modules/Microsoft.Maintenance/maintenanceConfigurations/.bicep/nested_roleAssignments.bicep:36]: updated

File: modules/Microsoft.Maintenance/maintenanceConfigurations/.test/common/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,70 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply modules/Microsoft.Maintenance/maintenanceConfigurations/.test/common/deploy.test.bicep]: Updated - via regex - soo.... there are a couple additional fixes

[the remark this answers is not in the document]
Comment (eriqua) [reply modules/Microsoft.Maintenance/maintenanceConfigurations/.test/common/deploy.test.bicep]: Sounds good, in that case probably we should also give the module readme generation a spin 🎇

[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply modules/Microsoft.Maintenance/maintenanceConfigurations/.test/common/deploy.test.bicep]: On it

[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply modules/Microsoft.Maintenance/maintenanceConfigurations/.test/common/deploy.test.bicep]: done

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**